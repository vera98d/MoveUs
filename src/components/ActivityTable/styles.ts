import styled from "styled-components";

export const TableWrapper = styled.div`
  width: 80vh;
  height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const ExerciseTable = styled.div`
  display: grid;
  text-align: center;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: [row1-start] 15%;
  column-rule: 1px solid red;
  grid-auto-rows: 12.14%;
  border: 1px solid black;
  width: 100%;
  height: 50vh;
`;

export const ExerciseCell = styled.p`
  padding: 8px 4px;
  border-left: 1px solid black;
  border-bottom: 1px solid black;
`;

export const PaginationWrapper = styled.div`
  width: 80%;
  display: flex;
  justify-content: space-around;
`;

export const PaginationList = styled.ul`
  display: flex;
  list-style-type: none;
  
  .linkStyle {
    text-decoration: none;
    color: inherit;
  }
`;

export const ListElement = styled.li`
`;
