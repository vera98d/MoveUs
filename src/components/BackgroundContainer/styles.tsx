import styled from "styled-components";

const BackgroundContainer = styled.div`
    width: 100%;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 2em 3em;
    background: ${(props) => props.theme.colors.background};
`;

export default BackgroundContainer;