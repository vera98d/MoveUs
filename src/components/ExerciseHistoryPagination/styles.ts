import styled from "styled-components";

export const PaginationList = styled.ul`
  display: flex;
  list-style-type: none;
  
  .linkStyle {
    text-decoration: none;
    color: inherit;
  }
`;

export const ListElement = styled.li`
`;
