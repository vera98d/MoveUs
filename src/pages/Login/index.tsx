import React from "react";
import { SubmitHandler, useForm } from "react-hook-form";
import { Link } from "react-router-dom";

type FormFields = {
    userName: string;
    password: string;
}

const Login = () => {
    const { register, handleSubmit, formState: { errors } } = useForm<FormFields>();
    const onSubmit: SubmitHandler<FormFields> = (data) => console.log(data, errors)
    const onSubmitError: SubmitHandler<any> = (data) => console.log(data, errors)

    return (
        <div className="login">
            <form onSubmit={handleSubmit(onSubmit, onSubmitError)}>
               <input
          type="text"
          {...register('userName', { required: true, minLength: 3, maxLength: 20 })}
          placeholder="Put your user name"
        />        
        {errors.userName?.type === 'required' && "Please put your user name"}
        {errors.userName?.type === 'minLength' && "Use at least 3 characters"}
        {errors.userName?.type === 'maxLength' && "You can use 20 characters at most"}
   
                <input
          type="password"
          {...register('password', { required: true, minLength: 8, })}
          placeholder="Put your password"
                />
                <button>Login</button>
                <div>
          Don't have an account?{" "}
          <Link to="/register">Register</Link> now.
        </div>
            </form>
        </div>
    )
}

export default Login;