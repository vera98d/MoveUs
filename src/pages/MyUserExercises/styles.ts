import styled from "styled-components";

export const BackgroundContainer = styled.div`
  width: 100%;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 2em 3em;
  background: ${(props) => props.theme.colors.background};
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  max-width: 1320px;
  width: 100%;
  margin: 0 auto;
  padding: 0 10px;
  
  .UserInformationContainer {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 45px;
  }
  
  .setPages {
    margin-bottom: 50px;
  }
  
  h3 {
    margin-bottom: 30px;
  }
  
  .groupWrapper {
    width: 30%;
    display: flex;
    align-items: center;
    flex-direction: column;
  }
  
  .usernameStyle {
    color: ${(props) => props.theme.colors.secondaryUi};
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    font-size: 18px;
  }
`;

export const LineWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px 0;
  
  .lineWrapperContent {
    margin: 20px 0;
    width: 100%;
    display: flex;
    justify-content: space-between;
  }
`;

export const GroupContainer = styled.div`
  margin-bottom: 60px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  
  .verticalLine {
    padding: 0 15px;
  }
`;
