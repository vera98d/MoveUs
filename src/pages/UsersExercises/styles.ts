import styled from "styled-components";

export const Img = styled.img`
  border-radius: 50%;
  width: 70px;
  height: 70px;
  margin: 0 30px 0 0;
`;

export const BackgroundContainer = styled.div`
  width: 100%;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 2em 3em;
  background: ${(props) => props.theme.colors.background};
`;

export const Header = styled.h1`
  font-size: ${(props) => props.theme.fontSize.header};
  color: ${(props) => props.theme.colors.secondaryUi};
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  margin-bottom: 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  max-width: 1320px;
  width: 100%;
  margin: 0 auto;
  padding: 0 10px;

  .UserInformationContainer {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  h3 {
    margin-bottom: 30px;
  }

  .groupWrapper {
    width: 30%;
    display: flex;
    align-items: center;
    flex-direction: column;
  }

  .usernameStyle {
    color: ${(props) => props.theme.colors.secondaryUi};
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    font-size: 24px;
    font-weight: bold;
  }
`;

export const LineWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px 0;
`;
