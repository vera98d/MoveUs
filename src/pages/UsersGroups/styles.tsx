import styled from "styled-components";

interface TileProps {
  img?: string;
}

export const Tile = styled.div<TileProps>`
    width: 130px;
    height: 130px;
    position: relative ;    
    background: ${(props) => (props.img ? `url(${props.img})` : props.theme.colors.white)};
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    opacity: 80%;
    border: ${(props) => props.theme.border};
    border-radius: 25px;
    cursor: pointer;

    &::before{
        content: "";
        display: block;
        position: absolute;
        top: 0;
        left:0;
        width:100%;
        height: 100%;
        background: rgba(255,255,255,0.2);
        border-radius: 25px;
        backdrop-filter: blur(.5px);
    }
`;

export const TileContent = styled.p`
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: ${(props) => props.theme.fontSize.tiles};
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    color: ${(props) => props.theme.colors.secondaryUi};
    position: relative ;
    width: 100%;
    height: 100%;
    padding: .5em;
    z-index: 2;
`;
export const TileContainer = styled.div`
    max-width: 1100px;
    width: 100%;
    display: flex;
    align-items:center;
    justify-content: center;
    flex-wrap: wrap;
    gap: 1.5em;
    margin: 1.5em 0;
    text-align: center;
    
    &>a{
        text-decoration: none;
    }
`;
