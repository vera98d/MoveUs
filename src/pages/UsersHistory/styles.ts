import styled from "styled-components";

export const Img = styled.img`
  border-radius: 50%;
  width: 75px;
  height: 75px;
  margin: 0 30px 0 0;
`;

export const Test = styled.div`
`;

export const Header = styled.header`
  height: 130px;
  background-color: aqua;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  max-width: 1320px;
  width: 100%;
  margin: 0 auto;
  padding: 0 10px;
  
  .UserInformationContainer {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 45px;
  }
    
  h3 {
    margin-bottom: 30px;
  }
  
  .groupWrapper {
    width: 30%;
    display: flex;
    align-items: center;
    flex-direction: column;
  }
`;

export const LineWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px 0;
`;
