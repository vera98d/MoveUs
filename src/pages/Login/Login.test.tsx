import { render, screen } from "@testing-library/react";
import {BrowserRouter as Router} from 'react-router-dom';
import { ThemeProvider } from "styled-components";
import { theme } from "../../styles";
import Login from "./index";

jest.mock("firebase/auth", () => ({
  getAuth: () => {}
}));

jest.mock("react-firebase-hooks/auth", () => ({
  useAuthState: () => ([])
}));

describe("Login", () => {
    test("matches snapshot", async () => {
      const { container } = render(
        <ThemeProvider theme={theme}>
          <Router>
            <Login />
          </Router>
        </ThemeProvider>
      );
      expect(container).toMatchSnapshot();
    });

    test('render email input', () => {
      render(
      <ThemeProvider theme={theme}>
        <Router>
          <Login />
        </Router>
      </ThemeProvider>
      );
     
      const inputEl = screen.getByTestId("email-input");
      expect(inputEl).toBeInTheDocument();
      expect(inputEl).toHaveAttribute("type", "email");
    });

    test('render password input', () => {
      render(
        <ThemeProvider theme={theme}>
          <Router>
            <Login />
          </Router>
        </ThemeProvider>
      );
       
      const inputEl = screen.getByTestId("password-input");
      expect(inputEl).toBeInTheDocument();
      expect(inputEl).toHaveAttribute("type", "password");
    });
});