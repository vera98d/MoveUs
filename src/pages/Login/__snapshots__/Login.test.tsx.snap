// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ llUgnK"
  >
    <div
      class="sc-dkzDqf dBGSmS"
    >
      <form
        class="sc-hKMtZM dkUaUq"
      >
        <div
          class="sc-eCYdqJ Kiuuf"
        >
          <p
            class="sc-jSMfEi lmaPab"
          >
            Enter your email
          </p>
          <input
            autocomplete="email"
            class="sc-gKXOVf gyVkUA"
            data-testid="email-input"
            id="email-input"
            name="email"
            placeholder="Email"
            type="email"
          />
          <span
            class="sc-ftvSup dQGNRI"
          />
        </div>
        <div
          class="sc-eCYdqJ Kiuuf"
        >
          <p
            class="sc-jSMfEi lmaPab"
          >
            Enter your password
          </p>
          <input
            class="sc-gKXOVf gyVkUA"
            data-testid="password-input"
            id="password-input"
            name="password"
            placeholder="Enter your password"
            type="password"
          />
        </div>
        <button
          class="sc-iBkjds gMQUUp"
        >
          Login
        </button>
        <div
          class="sc-papXJ eMOwUv"
        >
          Not a member yet?
           
          <a
            class="sc-jqUVSM eBkEai"
            href="/team-jo-project-4/register"
          >
            Register now
          </a>
           
        </div>
      </form>
    </div>
  </div>
</div>
`;
